File: Easy/Java/2367. Number of Arithmetic Triplets.java
class Solution {
    public int arithmeticTriplets(int[] nums, int diff) {
        int count = 0;
        Set<Integer> set = new HashSet();
        
        for(int n : nums){
            if(set.contains(n - diff) && set.contains(n - diff * 2))
                count++;
            set.add(n);
        }

        return count;
    }
}

File: Easy/Python/2367. Number of Arithmetic Triplets.py
class Solution(object):
    def arithmeticTriplets(self, nums, diff):
        count = 0
        mySet = set()
        
        for n in nums:
            if (n - diff) in mySet and (n - diff * 2) in mySet:
                count += 1
            mySet.add(n)

        return count
